keras_cv/layers/segmentation/segmentation_head.py

```diff
@@ -0,0 +1,159 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import tensorflow as tf
+from tensorflow import keras
+from tensorflow.keras import layers
+
+
+@keras.utils.register_keras_serializable(package="keras_cv")
+class SegmentationHead(layers.Layer):
```

Comment:
I think this probably doesn't make sense as a standalone item in the `layers` API, as it may not be the right choice for some other segmentation models.

It's also a lot of logic for something that could (and probably _should_) be expressed as just a `keras.Sequential`.

Let's define this head inside `deeplab_v3.py` for now and just make it a sequential model (this also gets rid of the dictionary input issue)

Reply:
I think this whole model can be expressed as:

```
keras.Sequential([
        keras.layers.Conv2D(
                    filters=256,
                    kernel_size=1,
                    padding="same",
                    use_bias=False,
                ),
        keras.layers.BatchNormalization(),
        keras.layers.Conv2D(
            filters=num_classes,
            kernel_size=1,
            use_bias=False,
            padding="same",
            activation="relu",
            # Force the dtype of the classification head to float32 to avoid the
            # NAN loss issue when used with mixed precision API.
            dtype=tf.float32,
        )

])
```

(But do double check me on that)